contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -2384,13 +2387,90 @@ static void bbf_ProcessUtility(PlannedStmt *pstmt,
 
 							if (strcmp(defel->defname, "default_database") == 0)
 								login_options = lappend(login_options, defel);
+							else if (strcmp(defel->defname, "name_location") == 0)
+							{
+								int location = defel->location;
+								orig_loginname = extract_identifier(queryString + location);
+								login_options = lappend(login_options, defel);
+							}
+							else if (strcmp(defel->defname, "from_windows") == 0)
+							{
+								from_windows = true;
+								login_options = lappend(login_options, defel);
+							}
 						}
 
 						foreach(option, login_options)
 						{
 							stmt->options = list_delete_ptr(stmt->options,
 															lfirst(option));
 						}
+
+						if (orig_loginname)
+						{
+							login_options = lappend(login_options, 
+													makeDefElem("original_login_name",
+																(Node *) makeString(orig_loginname),
+																-1));
+						}
+
+						if (from_windows && orig_loginname)
```

Comment:
I feel orig_loginname don't have to bound with from windows or it should be aways be available

Reply:
Yes, we are storing the orig_loginname in the previous block. For all types of login (password based and windows login), original login name will be stored.

Reply:
I think what Di wants to say here is that, few checks are also applicable to password based auth. So for all such checks, we need to put outside of if (from_windows) block. And whatever is applicable to only windows login should stay there.